### ISSUE TITLE
Should use  use path joining as the path separator,  instead of hard-codes / .

### ISSUE THREAD

**Issue by golanguage** [nabijaczleweli/rust-embed-resource#71]
[Link ](https://github.com/rust-lang/cargo/issues/13919#issuecomment-2462587860)

Win10 system, build a project which depends on gpui, 

after` rustup overfide set nightly` and `rustup update` to 1.84-nighgly
On pwsh 7.4.6, `cargo build`  panic:
```
error: failed to run custom build command for `gpui v0.1.0 (C:\proj\rust\zed\crates\gpui)`
  cargo:rerun-if-changed=resources/windows/gpui.rc
  Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384
  Copyright (C) Microsoft Corporation.  All rights reserved.

  fatal error RC1109: error creating \\?\C:\myproj\rust\gui\gpuimg\target\debug\build\gpui-a349b6bdb1d76471\out/gpui.lib

  --- stderr
  thread 'main' panicked at C:\rust\cargo\registry\src\rsproxy.cn-0dccff568467c15b\embed-resource-2.5.0\src\windows_msvc.rs:39:13:
  RC.EXE failed to compile specified resource file
```

A rust lang member said,  it (embed-resource) does not use path joining and instead hard-codes / as the path separator, which is wrong for \\?\ paths. [here](https://github.com/nabijaczleweli/rust-embed-resource/blob/0156d82afb13a2edbc6e0a5aad39b9317c204360/src/windows_msvc.rs#L31) and [here](https://github.com/nabijaczleweli/rust-embed-resource/blob/0156d82afb13a2edbc6e0a5aad39b9317c204360/src/windows_not_msvc.rs#L26)

On powershell 6 ,`cargo `is working correctly with 1.84 nightly.

**Comment by nabijaczleweli** [comment]
`/` is still not a valid character in a filename, and it's a valid directory separator universally (except in this one specific edge-case on Win32), so cargo needs to apply the same fixup for all the paths it reads from `cargo:rustc-link-arg-*=` anyway to fix the entire extant corpus that does `println!(cargo:rustc-link-arg-bins=$OUT_DIR/whatever)` or `...=$(realpath .)/whatever`.

As for building it here: yeah, sure, that's marginally a bug since we give it to other programs. Can you try the current `master` branch (at least 35a9ac6a7ba254ab706b6a053a35600fc8aed748)?

**Comment by nabijaczleweli** [comment]
If the downstream you're building is still on embed-resource 2, use the `v2.5-stable` branch (at least af3fed9a44f27f6995de55e1b39205a418568a60).

**Comment by nabijaczleweli** [comment]
So, you should have something like
```toml
gpui = { git = "https://github.com/zed-industries/zed" }
```
in your Cargo.toml (and this fails to build for you) If you replace it with
```toml
gpui = { git = "https://github.com/nabijaczleweli/zed" }
```
then you'll use https://github.com/nabijaczleweli/zed with https://github.com/nabijaczleweli/zed/commit/c49760a6645d37a1c9d11f2525c2f46eb4a30a52 that will use the `v2.5-stable` branch off git. Can you try building this, and seeing if it fixes the build?

**Comment by golanguage** [comment]
OK.
Now on pwsh 7.4.6 I get
```
...
...
   Compiling embed-resource v2.5.1 (https://github.com/nabijaczleweli/rust-embed-resource?branch=v2.5-stable#af3fed9a)
   Compiling imagesize v0.13.0
   Compiling roxmltree v0.20.0
   Compiling collections v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling derive_refineable v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling http_client v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling util v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling gpui v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling sum_tree v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling gpui_macros v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling usvg v0.44.0
   Compiling refineable v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling semantic_version v0.1.0 (https://github.com/nabijaczleweli/zed#c49760a6)
   Compiling resvg v0.44.0
   Compiling gpuimg v0.1.0 (C:\myproj\rust\gui\gpuimg)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1m 38s
     Running `target\debug\gpuimg.exe`
'''
Great!

**Comment by nabijaczleweli** [comment]
Cool, released in [`v3.0.1`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v3.0.1), backported as [`v2.5.1`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v2.5.1) and [`v1.6.14`](https://github.com/nabijaczleweli/rust-embed-resource/releases/tag/v1.6.14).
